provider.go

```diff
@@ -72,6 +89,8 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	if m := awsUrlRegexp.FindStringSubmatch(parsedUrl.Hostname()); m != nil {
 		log.Printf("[INFO] Using AWS: %+v", m[1])
 		opts = append(opts, elastic.SetHttpClient(awsHttpClient(m[1], d)), elastic.SetSniff(false))
+	} else {
+		opts = append(opts, elastic.SetHttpClient(tlsHttpClient(d)), elastic.SetSniff(false))
```

Comment:
Would we always want to create this http client even if we don't have a cacert_file? Should this be `else if (d.Get("insecure") || d.Get("cacert_file"))` ?

Reply:
Sounds good. I'll update my PR.

Reply:
Thanks, don't forget to squash please!

Reply:
Done!